cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -50,28 +102,409 @@ CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
+  <logging-monitoring-system-services-logs>` (such as the *Metrics Service*) that run in YARN. 
+- The ``logback.xml`` configures the edge node-located :ref:`CDAP System Services
+  <logging-monitoring-system-services-logs>`, such as the *App Fabric* or *Data Service*.
+- Both of these files are, in Distributed CDAP, located in ``/etc/cdap/conf``.
+- In the case of :ref:`Standalone CDAP <modes-data-application-platform>`, as containers
+  are replaced with Java threads, only a single file (``logback.xml``, located in
+  ``<cdap-sdk-home>/conf``) is used, replacing the ``logback.xml`` and
+  ``logback-container.xml`` of Distributed CDAP.
+- The ``cdap-site.xml`` controls the CDAP Log Pipeline for CDAP System Services, and the
+  reading, buffering, and memory allocated.
+- Custom "logback" files, if present in a specified directory, trigger the creation of
+  custom logging pipelines, with a pipeline created for each custom file.
+
+.. _logging-monitoring-custom-logging-pipeline:
+
+Custom Logging Pipeline
+=======================
+For a custom logging pipeline, you would create and configure a Logback XML file,
+configuring loggers, appenders, and properties based on your requirements, and place the
+file at the path specified in the ``cdap-site.xml`` file by the property
+``log.process.pipeline.config.dir`` of the ``cdap-site.xml`` file.
+
+Each custom pipeline requires a unique name, which is used for persisting the data and the
+retrieving of metadata. Properties controlling the pipeline (the
+``log.process.pipeline.*`` properties) are described :ref:`above
+<custom-log-pipeline-logging-configuration>`.
+
+For every XML file in the ``log.process.pipeline.config.dir`` directory, a separate logging pipeline is created, providing
+isolation from other logging pipelines. As they are separate Kafka consumers and
+processes, each pipeline is independent of |---| and doesn't affect the performance of
+|---| other logging pipelines. Though CDAP has been tested with multiple logging pipelines
+and appenders, the fewer of each that are specified will provide better performance.
+
+.. _logging-monitoring-custom-logging-configuration:
+
+Custom Logging Pipeline Configuration Pipelines
+-----------------------------------------------
+CDAP looks for Logback "logback" files located in a directory as set by the property
+``log.process.pipeline.config.dir`` in the :ref:`cdap-site.xml
+<appendix-cdap-default-logging>` file. In the default configuration, this is:
+  
+- For **Standalone CDAP:** ``<cdap-sdk-home>/ext/logging/config``
+- For **Distributed CDAP:** ``/opt/cdap/master/ext/logging/config``
+
+.. _logging-monitoring-custom-logging-example:
+
+Example "logback" File for a Custom Logging Pipeline
+----------------------------------------------------
+.. highlight:: xml
+
+Here is an example "logback" file, using two appenders (``STDOUT`` and
+``rollingAppender``). It would be saved with the file extension ``.xml`` to a location as
```

Comment:
"would be saved" sounds strange. Better: This file must be located in the location ... with a file extension of ".xml"